docs/api-reference/automatic-compaction-api.md

```diff
@@ -23,69 +23,762 @@ sidebar_label: Automatic compaction
+                "chatHandlerTimeout": null,
+                "chatHandlerNumRetries": null,
+                "maxNumSegmentsToMerge": null,
+                "totalNumMergeTasks": null,
+                "maxColumnsToMerge": null,
+                "type": "index_parallel",
+                "forceGuaranteedRollup": false
+            },
+            "granularitySpec": {
+                "segmentGranularity": "DAY",
+                "queryGranularity": null,
+                "rollup": null
+            },
+            "dimensionsSpec": null,
+            "metricsSpec": null,
+            "transformSpec": null,
+            "ioConfig": null,
+            "taskContext": null
+        },
+        "auditInfo": {
+            "author": "",
+            "comment": "",
+            "ip": "127.0.0.1"
+        },
+        "auditTime": "2023-07-31T18:16:16.362Z"
+    }
+]
+```
+</details>
+
+## Automatic compaction status
+
+### Get segments awaiting compaction
+
+Returns the total size of segments awaiting compaction for the given datasource. The specified datasource must have automatic compaction enabled.
+
+#### URL
+
+<code class="getAPI">GET</code> <code>/druid/coordinator/v1/compaction/progress?dataSource=:dataSource</code>
+
+#### Query parameter
+* `dataSource` (required)
+  * Type: String
+  * Name of the datasource for this status information
+
+#### Responses
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--200 SUCCESS-->
+
+*Successfully retrieved segment size awaiting compaction* 
+
+<!--404 NOT FOUND-->
+
+*Unknown datasource name or datasource does not have automatic compaction enabled* 
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+---
+
+#### Sample request
+
+The following example retrieves the remaining segments to be compacted for datasource `wikipedia_hour`.
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--cURL-->
+
+```shell
+curl "http://ROUTER_IP:ROUTER_PORT/druid/coordinator/v1/compaction/progress?dataSource=wikipedia_hour"
+```
+
+<!--HTTP-->
+
+```HTTP
+GET /druid/coordinator/v1/compaction/progress?dataSource=wikipedia_hour HTTP/1.1
+Host: http://ROUTER_IP:ROUTER_PORT
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+#### Sample response
+
+<details>
+  <summary>Click to show sample response</summary>
+
+```json
+{
+    "remainingSegmentSize": 7615837
+}
+```
+</details>
+
+
+### Get compaction status and statistics
+
+Retrieves the status and statistics from the automatic compaction run of all datasources which have automatic compaction enabled in the latest run. Returns a list of `latestStatus` objects. Each `latestStatus` represents the status for a datasource with automatic compaction enabled.
+
+The `latestStatus` object has the following keys:
+* `dataSource`: Name of the datasource for this status information.
+* `scheduleStatus`: Automatic compaction scheduling status. Possible values are `NOT_ENABLED` and `RUNNING`. Returns `RUNNING ` if the dataSource has an active automatic compaction config submitted. Otherwise, returns `NOT_ENABLED`.
```

Comment:
```suggestion
* `scheduleStatus`: Automatic compaction scheduling status. Possible values are `NOT_ENABLED` and `RUNNING`. Returns `RUNNING ` if the datasource has an active automatic compaction configuration submitted. Otherwise, returns `NOT_ENABLED`.
```